app/controllers/registrations_controller.rb

```diff
@@ -7,4 +7,15 @@ def new
       redirect_to root_path, alert: 'Sign up is only allowed via invitation.'
     end
   end
+
+  def update
+    if params[:student][:email] != current_student.email && current_student.valid_password?(params[:student][:current_password])
+      begin
+        current_student.update_close_email(params[:student][:email])
+      rescue Exception => e
```

Comment:
In this case, handling it appropriately might mean emailing Debbie to let her know the student's name, former email address, and new email address. Or you might not want to rescue it, and instead just get notified through Bugsnag (or whatever you're using) so you can deal with it.